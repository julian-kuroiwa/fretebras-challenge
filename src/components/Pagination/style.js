import styled from 'styled-components';

export const Container = styled.div`
  margin-top: 40px;

  ul {
    :none ;
    display: flex;
    justify-content: center;

    > li {
      & + li {
        margin-left: 20px;
      }

      button {
        height: 40px;
        width: 40px;
        border-radius: 50%;
        border: 0;
        background: transparent;
        color: #c53b31;

        &.active {
          background: #c53b31;
          color: #fff;
        }
      }
    }
  }
`;
