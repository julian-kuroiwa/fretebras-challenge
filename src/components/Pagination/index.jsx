import React from 'react';

import { Container } from './style';

const Pagination = () => {
  return (
    <Container>
      <nav>
        <ul>
          <li>
            <a href="#">1</a>
          </li>
          <li>
            <a href="#">2</a>
          </li>
          <li>
            <a href="#">3</a>
          </li>
          <li>
            <a href="#">4</a>
          </li>
          <li>
            <a href="#">5</a>
          </li>
        </ul>
      </nav>
    </Container>
  );
};

export default Pagination;
