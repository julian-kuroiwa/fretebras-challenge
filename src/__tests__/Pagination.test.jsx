import { shallow, mount } from 'enzyme';
import React from 'react';

import Pagination from '../components/Pagination';

describe('Pagination', () => {
  it('should match snapshot', () => {
    const component = shallow(
      <Pagination totalPages={5} startsAt={0} perPage={50} />,
    );

    expect(component).toMatchSnapshot();
  });

  it('should render with pages index', () => {
    const component = shallow(
      <Pagination totalPages={5} startsAt={0} perPage={50} />,
    );

    expect(component.find('ul').children().length).toBe(5);
  });
});
