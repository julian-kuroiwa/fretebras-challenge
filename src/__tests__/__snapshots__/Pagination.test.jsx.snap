// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should match snapshot 1`] = `
<styled.div>
  <nav>
    <ul>
      <li
        key="0"
      >
        <button
          className="active"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
      </li>
      <li
        key="1"
      >
        <button
          className=""
          onClick={[Function]}
          type="button"
        >
          2
        </button>
      </li>
      <li
        key="2"
      >
        <button
          className=""
          onClick={[Function]}
          type="button"
        >
          3
        </button>
      </li>
      <li
        key="3"
      >
        <button
          className=""
          onClick={[Function]}
          type="button"
        >
          4
        </button>
      </li>
      <li
        key="4"
      >
        <button
          className=""
          onClick={[Function]}
          type="button"
        >
          5
        </button>
      </li>
    </ul>
  </nav>
</styled.div>
`;
